tools/generator/dfg/stm32/stm.py

```diff
@@ -89,6 +89,71 @@ def getGpioRemapForModuleConfig(module, config):
         mmm['mapping'] = stm32f1_gpio_remap[module]['mapping'][int(config)]
     return mmm
 
+stm32f3_dma_remap = \
+{
+    'dma1ch1': {
+        'tim17_ch1': 'tim17_up',
+        'tim17_up':  {'position': 12, 'clear': 1, 'set': 0},
+    },
+    'dma1ch2': {
+        'adc2':      {'position': 72, 'clear': 3, 'set': 2},
+        'i2c_tx':    {'position': 70, 'clear': 3, 'set': 1},
+        'spi_rx':    {'position': 64, 'clear': 3, 'set': 0}, # also 'set': 3
+    },
+    'dma1ch3': {
+        'dac1_ch1':  'tim6_up',
+        'i2c_rx':    {'position': 68, 'clear': 3, 'set': 1},
+        'spi_tx':    {'position': 66, 'clear': 3, 'set': 0}, # also 'set': 3
+        'tim16_ch1': 'tim16_up',
+        'tim16_up':  {'position': 11, 'clear': 1, 'set': 0},
+        'tim6_up':   {'position': 13, 'clear': 1, 'set': 1},
+    },
+    'dma1ch4': {
+        'adc2':      {'position': 72, 'clear': 3, 'set': 3},
+        'dac1_ch2':  'tim7_up',
+        'i2c_tx':    {'position': 70, 'clear': 3, 'set': 2},
+        'spi_rx':    {'position': 64, 'clear': 3, 'set': 1},
+        'tim7_up':   {'position': 14, 'clear': 1, 'set': 1},
+    },
+    'dma1ch5': {
+        'dac2_ch1':  'tim18_up',
+        'i2c_rx':    {'position': 68, 'clear': 3, 'set': 2},
+        'spi_tx':    {'position': 66, 'clear': 3, 'set': 1},
+        'tim18_up':  {'position': 15, 'clear': 1, 'set': 1},
+    },
+    'dma1ch6': {
+        'i2c_tx':    {'position': 70, 'clear': 3, 'set': 0}, # also 'set': 3
+        'spi_rx':    {'position': 64, 'clear': 3, 'set': 2},
+    },
+    'dma1ch7': {
+        'i2c_rx':    {'position': 68, 'clear': 3, 'set': 0}, # also 'set': 3
+        'spi_tx':    {'position': 66, 'clear': 3, 'set': 2},
+        'tim17_ch1': 'tim17_up',
+        'tim17_up':  {'position': 12, 'clear': 1, 'set': 1},
```

Comment:
The STM32 reference manual incredibly inconsistent. All F3 reference manuals say this:
<img width="617" alt="screen shot 2018-09-19 at 19 58 15" src="https://user-images.githubusercontent.com/163066/45772002-ee818b80-bc46-11e8-87cc-7b5f206c0150.png">
With the corresponding table entry:
![screen shot 2018-09-19 at 19 58 48](https://user-images.githubusercontent.com/163066/45772005-f0e3e580-bc46-11e8-872c-0e9242b96d1a.png)

But the F373 reference manual says for this:
![screen shot 2018-09-19 at 19 57 49](https://user-images.githubusercontent.com/163066/45771962-ca25af00-bc46-11e8-902a-22fe83babdce.png)
![screen shot 2018-09-19 at 19 58 08](https://user-images.githubusercontent.com/163066/45771979-dad62500-bc46-11e8-9ef6-671616a1dbcb.png)

This is a mistake, the table is the same for all F3 and the CubeMX data confirms this. ST needs to get their shit in order.